src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,171 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+        if (!m_healEffectDummy)
+            return false;
+
+        return true;
+    }
+
+    void CastPrayerOfMendingAura(Unit* caster, Unit* target, uint8 stack)
+    {
+        uint32 basePoints = caster->SpellHealingBonusDone(target, m_spellInfoHeal, m_healEffectDummy->CalcValue(caster), HEAL, m_healEffectDummy);
+        CustomSpellValues values;
+        values.AddSpellMod(SPELLVALUE_AURA_STACK, stack);
+        values.AddSpellMod(SPELLVALUE_BASE_POINT0, basePoints);
+        caster->CastCustomSpell(SPELL_PRIEST_PRAYER_OF_MENDING_AURA, values, target, TRIGGERED_FULL_MASK);
+    }
+
+protected:
+    SpellInfo const* m_spellInfoHeal;
+    SpellEffectInfo const* m_healEffectDummy;
+};
+
+// 33076 - Prayer of Mending
+class spell_pri_prayer_of_mending : public SpellScriptLoader
+{
+public:
+    spell_pri_prayer_of_mending() : SpellScriptLoader("spell_pri_prayer_of_mending") { }
+
+    class spell_pri_prayer_of_mending_SpellScript : public spell_pri_prayer_of_mending_SpellScriptBase
+    {
+        PrepareSpellScript(spell_pri_prayer_of_mending_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_AURA });
+        }
+
+        void HandleEffectDummy(SpellEffIndex effIndex)
+        {
+            SpellEffectInfo const *effectInfo = GetEffectInfo(effIndex);
+            if (!effectInfo)
+                return;
+
+            Unit *caster = GetCaster();
+            if (!caster)
+                return;
+
+            Unit *target = GetExplTargetUnit();
+            if (!target)
+                return;
+
+            CastPrayerOfMendingAura(caster, target, effectInfo->BasePoints);
+        }
+
+        void Register() override
+        {
+            OnEffectHit += SpellEffectFn(spell_pri_prayer_of_mending_SpellScript::HandleEffectDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_pri_prayer_of_mending_SpellScript();
+    }
+};
+
+// 41635 - Prayer of Mending (Aura) - SPELL_PRIEST_PRAYER_OF_MENDING_AURA
+class spell_pri_prayer_of_mending_aura : public SpellScriptLoader
+{
+    public:
+        spell_pri_prayer_of_mending_aura() : SpellScriptLoader("spell_pri_prayer_of_mending_aura") { }
+
+        class spell_pri_prayer_of_mending_aura_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_pri_prayer_of_mending_aura_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_HEAL });
+            }
+
+            void HandleHeal(ProcEventInfo & /*eventInfo*/)
+            {
+                // Caster: the player(priest) that casters the prayer of mending
+                // Owner: the player that currently has the prayer of mending aura on him
+                if (Unit* owner = GetUnitOwner())
```

Comment:
owner should never be null, otherwise this cannot proc. Caster however can be null